core/src/main/scala/chisel3/Module.scala

```diff
@@ -560,18 +562,22 @@ package experimental {
     /** Chisel2 code didn't require the IO(...) wrapper and would assign a Chisel type directly to
       * io, then do operations on it. This binds a Chisel type in-place (mutably) as an IO.
       */
-    protected def _bindIoInPlace(iodef: Data)(implicit sourceInfo: SourceInfo): Unit = {
+    protected def _bindIoInPlace(iodef: IsBindable)(implicit sourceInfo: SourceInfo): Unit = {
 
-      // Assign any signals (Chisel or chisel3) with Unspecified/Flipped directions to Output/Input
-      Module.assignCompatDir(iodef)
+      // Assign any signals (Chisel or chisel3) with Unspecified/Flipped directions to Output/Input.
+      // This is only required for Data, not all IsBindables in general.
```

Comment:
sorry for the late comment here, but why is it not required for IsBindables... how does `IO(SomeProperty)` work (vs `IO(Output(SomeProperty)`) without this helper function?

Reply:
I thought there was a thread somewhere on this PR, but I can't find it. If you look at what assignCompatDir does, it only cares about Element, which is always Data: https://github.com/chipsalliance/chisel/blob/13bd5c8605c883d7b0ab5741eace1b01d1bc23af/core/src/main/scala/chisel3/Module.scala#L125.

As I understand it, this is a compatibility thing coming from chisel2... shouldn't we just disallow `IO(SomeProperty)`? Isn't that the old way, which we don't want?

Reply:
This is no longer coming from chisel2. Basically we adopted the Chisel2 semantics in chisel3, which is for an IO that you didn't say the direction on, its as if you put "Output" on it. 

> Isn't that the old way, which we don't want?

No, this is chisel3 semantics now.

Ultimately we would want `Incoming` and `Outgoing` but those do not exist yet.

Reply:
I mean, yes it is only for "Element" but that is the interesting base case of Data I'd imagine you'd come up with a "PropertyElement". But maybe once you have that or Property Aggregates you can worry about this, and I agree there is nothing for the generic  IsBindable

Reply:
Ok yeah, I see. I just made a small test and this seems fine to me. If you say IO(SomeProperty), you'll get an error explaining the port direction was unspecified, which seems reasonable. I think it's appropriate for users of properties to have to specify directions using Input and Output explicitly, or Incoming and Outgoing if those exist. I don't think we need to adopt the chisel2 semantics for properties, since they're a new thing.

Reply:
yes, and this really only gets interesting when you start having Aggregates (it is annoying to have to specify the directions on aggregate types, but you can make up your own property aggregate rules)